### PR TITLE
feat(cmd/ndt7-client): support gathering profile info

### PR DESCRIPTION
**Pull request by bassosimone**
Useful to investigate https://github.com/m-lab/ndt7-client-go/issues/59.

---
This change is [Reviewable](https://reviewable.io/reviews/m-lab/ndt7-client-go/60)

### CODE REVIEW

**Review by robertodauria — CHANGES_REQUESTED** [review]
Just a few little things, then :lgtm: - thanks!

> Reviewed 4 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/m-lab/ndt7-client-go/60#-:-MbHJFEL578xUAeB8eN7:bkslnxj)* status: 1 change requests, 0 of 1 approvals obtained (waiting on @bassosimone)

___
*[cmd/ndt7-client/main.go, line 41 at r1](https://reviewable.io/reviews/m-lab/ndt7-client-go/60#-MbHIsOGAqwr6PuoDkYv:-MbHIsOHCSC25dkUHYS1:b9l63sz) ([raw file](https://github.com/m-lab/ndt7-client-go/blob/6de87b916e7a6ed17a33c21c603cda84302fdfb3/cmd/ndt7-client/main.go#L41)):*
> ```Go
> // download.
> //
> // The `-cpuprofile` flag defines the file where to write a CPU profile
> ```

`s/cpuprofile/profile/`
___
*[cmd/ndt7-client/main.go, line 120 at r1](https://reviewable.io/reviews/m-lab/ndt7-client-go/60#-MbHIxxi6vNBYcMlLS3k:-MbHIxxj636RU5dqD8xX:b2j9bvw) ([raw file](https://github.com/m-lab/ndt7-client-go/blob/6de87b916e7a6ed17a33c21c603cda84302fdfb3/cmd/ndt7-client/main.go#L120)):*
> ```Go
> var (
> 	flagProfile = flag.String("profile", "",
> 		"file where to store cpuprofile (see https://blog.golang.org/pprof)")
> ```

`s/cpuprofile/pprof profile/`
___
*[cmd/ndt7-client/main.go, line 284 at r1](https://reviewable.io/reviews/m-lab/ndt7-client-go/60#-MbHJ0Xs3_s2WLQQha7b:-MbHJ0Xs3_s2WLQQha7c:bo5a1gd) ([raw file](https://github.com/m-lab/ndt7-client-go/blob/6de87b916e7a6ed17a33c21c603cda84302fdfb3/cmd/ndt7-client/main.go#L284)):*
> ```Go
> 	flag.Parse()
> 
> 	if *flagProfile != "" {
> ```

Could we add a warning that enabling the profiler may impact performance?

File: cmd/ndt7-client/main.go [hunk 1]
```diff
@@ -40,6 +38,9 @@
 // but may be set to false on the command line to run only upload or only
 // download.
 //
+// The `-cpuprofile` flag defines the file where to write a CPU profile
```
Comment (bassosimone) [inline cmd/ndt7-client/main.go:41]:
```suggestion
// The `-profile` flag defines the file where to write a CPU profile
```

File: cmd/ndt7-client/main.go [hunk 2]
```diff
@@ -113,6 +116,9 @@ const (
 )
 
 var (
+	flagProfile = flag.String("profile", "",
+		"file where to store cpuprofile (see https://blog.golang.org/pprof)")
```
Comment (bassosimone) [inline cmd/ndt7-client/main.go:120]:
```suggestion
		"file where to store pprof profile (see https://blog.golang.org/pprof)")
```

File: cmd/ndt7-client/main.go [hunk 3]
```diff
@@ -274,6 +280,16 @@ var osExit = os.Exit
 
 func main() {
 	flag.Parse()
+
+	if *flagProfile != "" {
```
Comment (bassosimone) [inline cmd/ndt7-client/main.go:284]:
```suggestion
	if *flagProfile != "" {
		log.Printf("warning: using -profile will obviously reduce the performance")	
```

**Review by bassosimone — COMMENTED** [review]
applied the required changes

File: cmd/ndt7-client/main.go [hunk 4]
```diff
@@ -274,6 +280,17 @@ var osExit = os.Exit
...
+		log.Printf("warning: using -profile will obviously reduce the performance")
```
Comment (bassosimone) [inline cmd/ndt7-client/main.go:285]:
```suggestion
		log.Printf("warning: using -profile will reduce the performance")	
```

**Review by robertodauria — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/m-lab/ndt7-client-go/60)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @bassosimone)

**Pull request merged**